cloud/aws/templates/aws_oidc/bin/aws_cli.py

```diff
@@ -53,47 +54,70 @@ def restart_ecs_service(self):
     def wait_for_ecs_service_healthy(self):
         """
         Polls the CiviForm ECS service, waiting for the PRIMARY deployment to
-        have rolloutStatus of COMPLETED.
+        have rolloutStatus of COMPLETED. If the PRIMARY deployment ID ends up
+        being equal to the previous_deployment_id, then the deployment failed
+        and we've rolled back.
 
         Gives up after 60 tries, sleeps 30 seconds between each try.
         """
         print(
             "\nWaiting for CiviForm ECS service to become healthy.\n"
             f"Service URL: {self._get_url_of_ecs_service()}")
 
+        error_text = inspect.cleandoc(
+            """
+            go to the Service URL printed above and click on the logs tab.
+            More details at https://docs.civiform.us/it-manual/sre-playbook/terraform-deploy-system/terraform-aws-deployment#inspecting-logs
+            For debugging help, contact the CiviForm oncall: https://docs.civiform.us/governance-and-management/project-management/on-call-guide#on-call-responsibilities
+            """)
+
+        tracked_deployment_id = None
         tries = 60
         while True:
-            state = self._ecs_service_state()
+            info = self._ecs_service_state()
+            id = info["id"]
+            state = info["state"]
+            tracked_deployment_id = id if tracked_deployment_id is None else tracked_deployment_id
+
             if state == "COMPLETED":
+                if id != tracked_deployment_id:
```

Comment:
suggestion (non-blocking): IIUC, `id` here would be the id of the deployment we rolled back to, and `tracked_deployment_id` is the id of the deployment we were attempting? What do you think about renaming `tracked_deployment_id` to `current_deployment_id`? That feels more clear to me, but also clarity is subjective.